### PR TITLE
Implement new API stores & refactor App component

### PR DESCRIPTION
**Pull request by maartenvn**
This PR includes a new API system, implemented inside Pinia stores and a refactor of the App.vue component.

Remarks:
* As of right now the stores are defined and hydrated with data, but not yet used throughout the app.
* Because of the refactor of app.vue the anonymize button does no longer work. This will be fixed once the other components are using the Pinia stores.
  * *This isn't really an issue right now, as we are not yet merging with main, but something to keep in mind.*

Reference #746, #754

### CODE REVIEW

**Review by rien — CHANGES_REQUESTED** [review]
Looking good!

It seems like the PR #726 was merged in main after the `refactor` branch was split off. There were some changes added to the API there as well. Can you look to rebase or merge your PR's with main?

File: web/src/App.vue [hunk 1]
```diff
@@ -120,49 +120,56 @@
...
+        router.push(route);
+      }
+    };
+
+    // Hydrate all the stores (fetch all the data).
+    (async () => {
+      await useFileStore().hydrate();
+      await useKgramStore().hydrate();
+      await useMetadataStore().hydrate();
+      await usePairStore().hydrate();
```
[the remark this answers is not in the document]
Comment (maartenvn) [reply web/src/App.vue:158]: This is fixed in the next PR (#759) 😄

File: web/src/api/stores/settings.store.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { defineStore } from "pinia";
+import { ref } from "@vue/composition-api";
+/**
+ * Store containing global settings.
+ */
+export const useSettingsStore = defineStore("settings", () => {
+  // Is the names should be anonymized.
```
[the remark this answers is not in the document]
Comment (maartenvn) [reply web/src/api/stores/settings.store.ts:7]: Will apply this in the next PR on this branch (this store has been renamed in #759)

[the remark this answers is not in the document]
Comment (maartenvn) [reply web/src/api/stores/settings.store.ts:7]: Fixed in 92c79c22c96f33c891e23e09f6ab82e32105c669

**Review by rien — APPROVED** [review]

**Pull request merged**